### PR TITLE
Add schemas package

### PR DESCRIPTION
**Pull request by hand-dot**
- [x] Move existing processing to a separate package (pdfme/schemas)
    - [x] Text
    - [x] Barcode
    - [x] Image
- [x] Move some pdfme/common functions to pdfme/schemas
- [x] Confirm that the pdfme/schema package can be injected from the outside

---

ref: https://github.com/pdfme/pdfme/issues/245

### CODE REVIEW

**Comment by hand-dot** [comment]
Hi @peteward,

I apologize for the very large pull request. Essentially, this update accomplishes the following:

- Created a schemas package
- Moved rendering processing from common package to schemas package
- import rendering processes from schemas to generator/ui

Could you please review the changes before merging?

File: packages/schemas/src/index.ts
Comment (hand-dot) [inline packages/schemas/src/index.ts:1]: I believe you'll find it easier to understand if you trace the process from this file.

File: packages/generator/src/builtInRenderer.ts
Comment (hand-dot) [inline packages/generator/src/builtInRenderer.ts:1]: In the future, barcodes will be excluded from the built-in features and will instead be loaded as plugins on the user side as needed. (The same applies to the UI package.)

Comment (hand-dot) [reply packages/generator/src/builtInRenderer.ts:1]: The following objects are used for their respective purposes and locations:
- ui: Used in Viewer, Form, Designer (`packages/ui/src/builtInRenderer.ts`)
- pdf: Used in Generator (`packages/generator/src/builtInRenderer.ts`)
- propPanel: Used in Designer (`packages/ui/src/builtInPropPanel.ts`)

File: packages/schemas/src/text/uiRender.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    backgroundColor: getBackgroundColor(mode, value, schema),
+    border: 'none',
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: mapVerticalAlignToFlex(schema.verticalAlignment),
+    width: '100%',
+    height: '100%',
+  };
+  Object.assign(container.style, containerStyle);
+  rootElement.innerHTML = '';
```
Comment (peteward) [inline packages/schemas/src/text/uiRender.ts:93]: I see these files have changed from JSX, which feels a bit harder to read. Was it not possible to use JSX?

**Review by peteward — APPROVED** [review]
Great work - I understand the changes.
Keep moving forwards and any small things can be iterated 👍

File: packages/schemas/src/text/uiRender.ts (hunk 1 above)
Comment (hand-dot) [reply packages/schemas/src/text/uiRender.ts:93]: @peteward 
Indeed, JSX is more readable.

However, requiring users to write plugins in React would mean forcing them to unify the UI package and React versions, as well as pre-transpile JSX.

While React might not be an option, handling UI rendering through direct DOM manipulation is a clear responsibility. Overall, this approach is the simplest.

**Pull request merged**